Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -3533,16 +3546,17 @@ class BBQJIT {
                 conditionLocation = Location::fromGPR(m_scratchGPR);
             }
 
-            if (lhs.isTemp())
-                lhs = Value::pinned(lhs.type(), lhsLocation);
-            if (rhs.isTemp())
-                rhs = Value::pinned(rhs.type(), rhsLocation);
-
             // Kind of gross isel, but it should handle all use/def aliasing cases correctly.
-            emitMove(lhs, resultLocation);
+            if (lhs.isConst())
+                emitMoveConst(lhs, resultLocation);
+            else
+                emitMove(lhs.type(), lhsLocation, resultLocation);
             Jump ifZero = m_jit.branchTest32(inverted ? ResultCondition::Zero : ResultCondition::NonZero, conditionLocation.asGPR(), conditionLocation.asGPR());
             consume(condition);
-            emitMove(rhs, resultLocation);
+            if (rhs.isConst())
+                emitMoveConst(rhs, resultLocation);
+            else
+                emitMove(rhs.type(), rhsLocation, resultLocation);
```

Comment:
This is executed only when we didn't take `m_jit.branchTest32`. But is it safe?
For example, `emitMove` is doing locationOf, which performs `bind` op. But `bind` is not safe when it is done only in a partial path in the basic block since it modifies register bank information which is shared in this basic block.